### PR TITLE
Docs reorg

### PR DESCRIPTION
**Pull request by elena-kolevska**
| **Data types tutorial subsection**     | **New page**                             |
|----------------------------------------|------------------------------------------|
| Keys                                   | New page: Using Redis/The Redis keyspace |
| Strings                                | Redis Strings                            |
| Altering and querying the key space    | New page: Using Redis/The Redis keyspace |
| Key expiration                         | New page: Using Redis/The Redis keyspace |
| Lists                                  | Redis lists                              |
| First steps with Redis Lists           | Redis lists                              |
| Common use cases for lists             | Redis lists                              |
| Capped lists                           | Redis lists                              |
| Blocking operations on lists           | Redis lists                              |
| Automatic creation and removal of keys | Redis lists                              |
| Hashes                                 | Redis hashes                             |
| Sets                                   | Redis sets                               |
| Sorted sets                            | Redis sorted sets                        |
| Operating on ranges                    | Redis sorted sets                        |
| Lexicographical scores                 | Redis sorted sets                        |
| Updating the score: leader boards      | Redis sorted sets                        |
| Bitmaps                                | Redis bitmaps                            |
| HyperLogLogs                           | Redis HyperLogLog                        |
| Other notable features                 | Remove                                   |
| Learn more                             | Remove                                   |

### CODE REVIEW

File: docs/data-types/hyperloglogs.md [hunk 1]
```diff
@@ -6,10 +6,48 @@ description: >
...
-HyperLogLog is a data structure that estimates the cardinality of a set. As a probabilistic data structure, HyperLogLog trades perfect accuracy for efficient space utilization.
+HyperLogLog is a probabilistic data structure that estimates the cardinality of a set. As a probabilistic data structure, HyperLogLog trades perfect accuracy for efficient space utilization.
 
 The Redis HyperLogLog implementation uses up to 12 KB and provides a standard error of 0.81%.
 
+Counting unique items usually requires an amount of memory
+proportional to the number of items you want to count, because you need
+to remember the elements you have already seen in the past in order to avoid
+counting them multiple times. However there is a set of algorithms that trade
+memory for precision: you end with an estimated measure with a standard error,
```
Comment (SoulPancake) [inline docs/data-types/hyperloglogs.md:17]:
```suggestion
However, a set of algorithms exist that trade memory for precision: they yield an estimated measure with a standard error, which, in the case of the Redis implementation, is less than 1%.
```

This improves the readability here in my opinion

File: docs/data-types/lists.md [hunk 2]
```diff
@@ -12,6 +12,279 @@ Redis lists are frequently used to:
...
+whatever they are: social network updates, logs, or anything else.
+
+Redis allows us to use lists as a capped collection, only remembering the latest
+N items and discarding all the oldest items using the [`LTRIM`](/commands/ltrim) command.
+
+The [`LTRIM`](/commands/ltrim) command is similar to [`LRANGE`](/commands/lrange), but **instead of displaying the
+specified range of elements** it sets this range as the new list value. All
+the elements outside the given range are removed.
+
+An example will make it more clear:
```
Comment (SoulPancake) [inline docs/data-types/lists.md:148]:
```suggestion
An example will make it clearer:
```

File: docs/data-types/lists.md [hunk 3]
```diff
@@ -12,6 +12,279 @@ Redis lists are frequently used to:
...
+
+The above [`LTRIM`](/commands/ltrim) command tells Redis to take just list elements from index
+0 to 2, everything else will be discarded. This allows for a very simple but
+useful pattern: doing a List push operation + a List trim operation together
+in order to add a new element and discard elements exceeding a limit:
+
+    LPUSH mylist <some element>
+    LTRIM mylist 0 999
+
+The above combination adds a new element and takes only the 1000
```
Comment (SoulPancake) [inline docs/data-types/lists.md:167]:
```suggestion
The above combination adds a new element and keeps only the 1000
```

**Review by SoulPancake — APPROVED** [review]
Just finished going through the entire thing,
LAwesomeTM and overall an excellent resource!

File: docs/data-types/lists.md [hunk 4]
```diff
@@ -12,6 +12,279 @@ Redis lists are frequently used to:
...
+    > rpush mylist 1 2 3 4 5
+    (integer) 5
+    > ltrim mylist 0 2
+    OK
+    > lrange mylist 0 -1
+    1) "1"
+    2) "2"
+    3) "3"
+
+The above [`LTRIM`](/commands/ltrim) command tells Redis to take just list elements from index
```
Comment (elena-kolevska) [inline docs/data-types/lists.md:159]:
```suggestion
The above [`LTRIM`](/commands/ltrim) command tells Redis to keep just list elements from index
```

File: docs/data-types/hyperloglogs.md (hunk 1 above)
Comment (elena-kolevska) [reply docs/data-types/hyperloglogs.md:17]: I think:

```
However, a set of algorithms exist that trade memory for precision: they return an estimated measure with a standard error, which, in the case of the Redis implementation for HyperLogLog, is less than 1%.
```

Comment (elena-kolevska) [reply docs/data-types/hyperloglogs.md:17]: Also, you'd have to remove "However there is a set of algorithms that trade" from line number 16.

**Comment by elena-kolevska** [comment]
Thanks for the detailed review @SoulPancake ! Great job!

File: docs/data-types/hyperloglogs.md (hunk 1 above)
Comment (SoulPancake) [reply docs/data-types/hyperloglogs.md:17]: So sorry Accidentally missed involving the lines accurately

Comment (SoulPancake) [reply docs/data-types/hyperloglogs.md:17]: Should I create a fresh suggestion for this ? @elena-kolevska 
"However, a set of algorithms exist that trade memory for precision: they return an estimated measure with a standard error, which, in the case of the Redis implementation for HyperLogLog, is less than 1%."

Comment (SoulPancake) [reply docs/data-types/hyperloglogs.md:17]: Can I open a PR to your branch ? @elena-kolevska

File: docs/data-types/hyperloglogs.md [hunk 5]
```diff
@@ -6,10 +6,48 @@ description: >
...
-HyperLogLog is a data structure that estimates the cardinality of a set. As a probabilistic data structure, HyperLogLog trades perfect accuracy for efficient space utilization.
+HyperLogLog is a probabilistic data structure that estimates the cardinality of a set. As a probabilistic data structure, HyperLogLog trades perfect accuracy for efficient space utilization.
 
 The Redis HyperLogLog implementation uses up to 12 KB and provides a standard error of 0.81%.
 
+Counting unique items usually requires an amount of memory
+proportional to the number of items you want to count, because you need
+to remember the elements you have already seen in the past in order to avoid
+counting them multiple times. However, a set of algorithms exist that trade 
+memory for precision: they return an estimated measure with a standard error, which, in the case of the Redis implementation for HyperLogLog, is less than 1%.
```
Comment (elena-kolevska) [inline docs/data-types/hyperloglogs.md:17]:
```suggestion
memory for precision: they return an estimated measure with a standard error, 
```

File: docs/data-types/hyperloglogs.md [hunk 6]
```diff
@@ -6,10 +6,48 @@ description: >
...
+which in the case of the Redis implementation is less than 1%.  The
```
Comment (elena-kolevska) [inline docs/data-types/hyperloglogs.md:18]:
```suggestion
which, in the case of the Redis implementation for HyperLogLog, is less than 1%.  The
```

**Review by SoulPancake — APPROVED** [review]

**Pull request merged**